gateway/api/views.py

```diff
@@ -173,7 +193,15 @@ def run(self, request):
 
                 jobconfig = config_serializer.save()
 
-            program = save_program(serializer=serializer, request=request)
+            program_service = self.get_service_program_class()
+            try:
+                program = program_service.save(
+                    serializer=serializer,
+                    author=request.user,
+                    artifact=request.FILES.get("artifact"),
+                )
+            except InternalServerErrorException as exception:
+                return Response(exception, exception.http_code)
```

Comment:
Should we log exception and return string something like `Internal server error` instead of stack trace?

Reply:
Yeah, that's exactly what I'm doing there. I created a set of custom exceptions and `InternalServerErrorException` is one of them. [Here](https://github.com/Qiskit-Extensions/quantum-serverless/pull/1112/files#diff-75b075598929b18be555b0fd14f9a1ea4472aad00cf2ab36d85044d3b20e179eR61) what I'm doing is catching the exception, logging it internally as an error and raising our custom exception to hide information to the user.

Reply:
OOOOH! looks great!